pennylane/vqe/vqe.py

```diff
@@ -120,6 +120,35 @@ def __str__(self):
 
         return "\n+ ".join(terms)
 
+    @property
+    def is_diagonal(self):
+        r""" Checks if a Hamiltonian is diagonal in the computational basis.
+
+        Returns:
+            bool: ``True`` if the Hamiltonian is diagonal in the computational basis, ``False`` otherwise
+        """
+
+        diagonals = ["PauliZ", "Identity"]
+
+        non_diagonal_coeffs = []
+        non_diagonal_obs = []
+
+        for i, term in enumerate(self._ops):
+            gates = [term.name] if isinstance(term.name, str) else term.name
+            for g in gates:
+                if g not in diagonals:
+                    if gates in non_diagonal_obs:
+                        non_diagonal_coeffs[non_diagonal_obs.index(gates)] += self._coeffs[i]
+                    else:
+                        non_diagonal_obs.append(gates)
+                        non_diagonal_coeffs.append(self._coeffs[i])
+                    break
```

Comment:
@Lucaman99 it might be worth including line comments here explaining the logic, I find it a bit hard to follow at first read.